Quantum-Control-Applications/Superconducting/Single Flux Tunable Transmon/SSB mixer for readout/resonator_spec.py

```diff
@@ -0,0 +1,161 @@
+"""
+resonator_spec.py: performs the 1D and 2D (with flux amplitude sweep) resonator spectroscopy
+"""
+from qm.qua import *
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from qm import SimulationConfig, LoopbackInterface
+from configuration import *
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import signal
+
+##############################
+# Program-specific variables #
+##############################
+
+n_avg = 30  # Number of averaging loops
+
+cooldown_time = 2000 // 4  # Resonator cooldown time in clock cycles (4ns)
+flux_settle_time = 4000 // 4  # Flux settle time in clock cycles (4ns)
+
+# Frequency sweep in Hz
+f_min = 55e6
+f_max = 65e6
+df = 0.05e6
+freqs = np.arange(f_min, f_max + df / 2, df)  # +df/2 to add f_max to the scan
+# Flux amplitude sweep (as a prefactor of the flux amplitude)
+a_min = -1
+a_max = 0
+da = 0.01
+flux = np.arange(a_min, a_max + da / 2, da)  # +da/2 to add a_max to the scan
+
+###################
+# The QUA program #
+###################
+
+with program() as resonator_spec_1D:
+    n = declare(int)  # Averaging index
+    f = declare(int)  # Resonator frequency
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+
+    with for_(n, 0, n < n_avg, n + 1):
+        with for_(f, f_min, f <= f_max, f + df):  # Notice it's <= to include f_max (This is only for integers!)
+            # Adjust the flux line
+            play("const" * amp(0), "flux_line")
```

Comment:
amp(0)?

Reply:
This was used to set the amplitude of the flux line manually, before we implemented the 2D scan. I can remove it, but I left it here in case someone wants to change it just for the 1D scan...

Reply:
I think it looks a bit weird. I would remove it (and just comment it out when not needed)